src/Orleans/Telemetry/IMetricsWriter.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Orleans.Runtime
+{
+    /// <summary>Represents an object used for publishing metrics</summary>
+    /// <remarks>In a future iteration, this will probably be replaced by an IMetric interface that already contains the name of the metric,
+    /// so callers will not need to specify it in every method call</remarks>
+    public interface IMetricsWriter
+    {
+        /// <summary>Send a metric value to the registered telemetry consumers.</summary>
+        /// <param name="name">Metric name.</param>
+        /// <param name="value">Metric value.</param>
+        /// <param name="properties">Named string values you can use to classify and filter metrics.</param>
+        void TrackMetric(string name, double value, IDictionary<string, string> properties = null);
+
+        /// <summary>Send a metric value to the registered telemetry consumers.</summary>
+        /// <param name="name">Metric name.</param>
+        /// <param name="value">Metric value.</param>
+        /// <param name="properties">Named string values you can use to classify and filter metrics.</param>
+        void TrackMetric(string name, TimeSpan value, IDictionary<string, string> properties = null);
```

Comment:
I feel this should be `long` instead of `TimeSpan`

Reply:
nah it is from old APM methods in Logger. it is designed to track TimeSpan values. 

The old APM methods design is not very flexible. It only tracks double and TimeSpan values. I think we should improve it in a separate PR. Maybe borrow some pattern from MEL, something like TrackMetric<T>(string name, IDictionary<string, string>) would be more flexible.

Reply:
I definitely +1 on redesign metric api in a separate PR. the current API design is not good.